### DISCUSSION TITLE
How to visualize visible colors in CIE Lab?

### DISCUSSION THREAD

**Discussion by hminle** [colour-science/colour#971] [Q&A] [answered]
Hi Colour Team,

Is there a way to visualize visible colors in CIE Lab?
Like this example: https://en.wikipedia.org/wiki/File:Visible_gamut_within_CIELAB_color_space_D65_whitepoint_mesh.png

Thanks

**Comment by thomasmansencal** [comment, marked as answer]
Hello,

We don't have a direct way to visualise it with Matplotlib but the solid can be built with the `colour.volume.solid_RoschMacAdam` definition that produces CIE XYZ values that can then be converted to *CIE Lab*.

We are using it there for example: https://www.colour-science.org:8020/

![image](https://user-images.githubusercontent.com/99779/166126853-02601490-80ba-417f-882b-f440abadd5f8.png)

Note: I think the Axis labels are flipped, I need to do a pass on that!

**Reply by hminle** [reply]
Thank you.
But I think you guys can visualize visible colors on CIE xy. Then you guys can slightly change to make it work on CIE Lab, is that right?

**Reply by thomasmansencal** [reply]
Sorry, just to clarify, are you talking in 2D or 3D here?

**Reply by hminle** [reply]
Hi, it's 3D, CIE Lab in 3D

**Reply by thomasmansencal** [reply]
We have limited 3D support in Colour and visualising the Rosch-MacAdam solid is not implemented atm. Given the poor state of MPL with 3D, it might never be.

**Reply by hminle** [reply]
I got it. Thank you anyway

**Reply by hminle** [reply]
Hi @KelSolaar , on the above visualization that you posted from https://www.colour-science.org:8020/
Is there any way to make the 3D visible gamut colorful?

**Comment by hminle** [comment]
Hi @KelSolaar , just curious about the Rosch-MacAdam solid in CIE 1931 XYZ.
If we can visualize all visible colors in CIE XYZ, the values of them can go beyond the range 0-100 in the axis X and Z? 
Like this figure
<img width="773" alt="image" src="https://user-images.githubusercontent.com/16201681/173487712-5a10d64b-bbfb-47d0-af28-c3459b38571d.png">

**Reply by thomasmansencal** [reply]
Hey @hminle,

This is a great question, technically, ignoring photoreceptor bleaching and saturation, there is no limit, it is however useful to bound them for practical purposes. Here we use a series of square pulses whose bandwidth grows from 1 wavelengths to the entire set of wavelengths to produce the Equal Energy and Unity Illuminant E. If you assume that E is the maximum possible **relative** radiance, then this is the limit of the Rosch-MacAdam solid.

Cheers,

Thomas

**Reply by hminle** [reply]
Thank you, I got it.
